tools/generate_dnsplugins_postrefreshhook.sh

```diff
@@ -0,0 +1,35 @@
+#!/bin/bash
+# Generate the hooks/post-refresh file for all DNS plugins
+set -e
+
+DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+CERTBOT_DIR="$(dirname "${DIR}")"
+
+for PLUGIN_PATH in "${CERTBOT_DIR}"/certbot-dns-*; do
+  PLUGIN=$(basename "${PLUGIN_PATH}")
+  mkdir -p "${PLUGIN_PATH}/snap/hooks"
+  cat <<EOF > "${PLUGIN_PATH}/snap/hooks/post-refresh"
+#!/bin/sh -e
+# This file is generated by tools/generate_dnsplugins_postrefreshhook.sh and should not be edited manually.
+
+# get certbot version
+if [ ! -f "\$SNAP/certbot-shared/certbot-version.txt" ]; then
+  echo "No certbot version available; not doing version comparison check" >> "\$SNAP_DATA/debuglog"
+  exit 0
+fi
+cb_installed=\$(cat \$SNAP/certbot-shared/certbot-version.txt)
+
+# get required certbot version for plugin. certbot version must be at least the plugin's
+# version. note that this is not the required version in setup.py, but the version number itself.
+cb_required=\$(grep -oP "version = '\K.*(?=')" \$SNAP/setup.py)
+
+
+$SNAP/bin/python3 -c "import sys; from setuptools._vendor.packaging import version; sys.exit(1) if\
```

Comment:
It looks like it works regardless, but I think we should add a backslash before `$SNAP` here. If you want, you can change the `set -e` line at the top of the file to `set -eu` and `bash` would have errored out about this.

Also, now that we have our own copy of `packaging`, can we stop using the private, vendorized version in setuptools?

Reply:
> It looks like it works regardless, but I think we should add a backslash before $SNAP here. If you want, you can change the set -e line at the top of the file to set -eu and bash would have errored out about this.

Ah, missed one, thanks.

> Also, now that we have our own copy of packaging, can we stop using the private, vendorized version in setuptools?

I thought I'd changed that! I guess I only changed it in the local testing copy... This is why I wanted to do a single version first, glad we did that.